examples/layout_padded/layout_padded.cpp

```diff
@@ -0,0 +1,648 @@
+      return extents_;
+    }
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr std::array<index_type, extents_type::rank()>
+    strides() const noexcept
+    {
+      return inner_mapping_.strides();
+    }
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr index_type required_span_size() const noexcept
+    {
+      return inner_mapping_.required_span_size();
+    }
+
+    MDSPAN_TEMPLATE_REQUIRES(
+      class... Indices,
+      /* requires */ (sizeof...(Indices) == Extents::rank() &&
+        _MDSPAN_FOLD_AND(_MDSPAN_TRAIT(std::is_convertible, Indices, index_type) /*&& ...*/ ) &&
+	_MDSPAN_FOLD_AND(_MDSPAN_TRAIT(std::is_nothrow_constructible, index_type, Indices) /*&& ...*/)
+      )
+    )
+    MDSPAN_FORCE_INLINE_FUNCTION
+    constexpr size_t operator()(Indices... idxs) const noexcept {
+      return inner_mapping_(std::forward<Indices>(idxs)...);
+    }
+
+    MDSPAN_INLINE_FUNCTION static constexpr bool is_always_unique() noexcept { return true; }
+    MDSPAN_INLINE_FUNCTION static constexpr bool is_always_exhaustive() noexcept {
+      return false; // just like layout_stride::mapping
+    }
+    MDSPAN_INLINE_FUNCTION static constexpr bool is_always_strided() noexcept { return true; }
+
+    MDSPAN_INLINE_FUNCTION static constexpr bool is_unique() noexcept { return true; }
+    MDSPAN_INLINE_FUNCTION _MDSPAN_CONSTEXPR_14 bool is_exhaustive() const noexcept {
+      return inner_mapping_.extents() == extents_;
+    }
+    MDSPAN_INLINE_FUNCTION static constexpr bool is_strided() noexcept { return true; }
+
+    MDSPAN_INLINE_FUNCTION
+    constexpr index_type stride(rank_type r) const noexcept {
+      return inner_mapping_.stride(r);
+    }
+
+  private:
+    extents_type extents_;
+    inner_mapping_type inner_mapping_;
+  };
+};
+
+// layout_padded tests
+
+template<StorageOrder order,
+	 std::size_t overalignment_factor,
+	 class ExtentsType,
+	 class ExpectedInnerExtentsType>
+void test_layout_padded_one(std::vector<float>& storage,
+			    std::integral_constant<StorageOrder, order>,
+			    std::integral_constant<std::size_t, overalignment_factor>,
+			    ExtentsType e,
+			    ExpectedInnerExtentsType expected_inner_extents,
+			    const bool expect_exhaustive)
+{
+  using layout_type = layout_padded<order, overalignment_factor>;
+  using extents_type = ExtentsType;
+  using expected_inner_extents_type = ExpectedInnerExtentsType;
+
+  using mdspan_type = stdex::mdspan<float, extents_type, layout_type>;
+  using mapping_type = typename mdspan_type::mapping_type;
+  mapping_type mapping(e);
+
+  {
+    std::size_t product_of_extents = 1;
+    for(std::size_t r = 0; r < e.rank(); ++r) {
+      product_of_extents *= std::size_t(e.extent(r));
+    }
+    const size_t min_num_elements_storage = mapping.required_span_size();
+    assert(min_num_elements_storage >= product_of_extents);
+    if(storage.size() < min_num_elements_storage) {
+      storage.resize(min_num_elements_storage);
+    }
+  }
+
+  mdspan_type x(storage.data(), mapping);
```

Comment:
Why do we bother with an actual storage if we never actually access elements?

Reply:
@dalg24 It's UB to create an mdspan with invalid storage.  I just want to test that this compiles.